### ISSUE TITLE
New Addon: Public post import

### ISSUE THREAD

**Issue by MrPetovan** [friendica/friendica#6022]
This addon is a ripoff the "Post Import" feature in Mastodon. If you find a public post on a remote server you want to interact with, you just pop the post import addon, paste the remote post URL, and the addon takes care of adding the relevant public contact if it doesnt exist, then imports the post as a public posts, redirecting to the generated `/display` page.

Thanks to npze for the suggestion!

Related to #6021

**Comment by annando** [comment]
Technically this is not a problem and also could easily be done with Diaspora posts.

**Comment by MrPetovan** [comment]
Any pointer? I know post import is done in every single Protocol differently, but is there a common `Item::...` method call that takes care of everything around an item?

**Comment by annando** [comment]
The Item-class is only for storing the whole stuff. The fetching is part of the protocol. For AP one could mostly use the private function ```fetchMissingActivity```.

**Comment by AlfredSK** [comment]
Maybe we can simply use the search field. Enter the remote post URL, hit enter and the item is presented on a 'display' page.

**Comment by rabuzarus** [comment]
It would be really cool if we could extend it for friendica/friendica contacts and use the advantage of OWA  in some way. 
I don't know if der are some security issues but maybe something like:
* Bob is on Alice Friendica instance and is authenticated
* Bow leave a comment to a puplic post on Alices profile page
* Alices friendica instance redirects the post in the background to some endpoint at Bobs instance
* Bobs instance checks if Bob is Bob (`local_user()`) and imports the whole public thread from Alice
* Bobs instance adds Bobs comment to the imported thread
* Bobs instance sends an "OK" to Alices instance that everything went fine
* Alices instance imports the comment from Bob
* Alices instance updates Alices profile page though Ajax
* Bob see his comment while he visually never left Alices instance

Like I said maybe this approach has downsides. I just want to start some discussion to use advantages of friendicas ability to authenticate non connected friendica contacts

**Comment by MrPetovan** [comment]
This looks so complicated it hurts my eyes.

**Comment by hoergen** [comment]
> Maybe we can simply use the search field. Enter the remote post URL, hit enter and the item is presented on a 'display' page.

I won't suggest to use the search field, because then you can't search for URLs any more (what is possible at the moment)

But at the moment I don't have a good idea how to do it, without copyNpasting it in a special "import form".

**Comment by AlfredSK** [comment]
Aha. What URLs can you search for? Didn't know that is possible.

**Comment by tobiasd** [comment]
> Aha. What URLs can you search for? Didn't know that is possible.

You can just search for any URL you want and then all postings containing the URL will be shown. Just like with any other word.

**Comment by rabuzarus** [comment]
For the user or the developer?
To have the possibility to comment on public posts on a foreign profile page is much more easier  for the user than going to his/her own instance, copy a link to import a post, to comment and to visit the foreign profile again if there are any other interesting posts.

BTW: with a different concept it is already possible in friendica. When two friendica contacts are connected one can visit the others profile and can comment.
It would be nice if such a feature would be also available if the two contacts aren't connected to each other and the posts are public.

**Comment by AlfredSK** [comment]
Ok. I guess the full text search has to be enabled for that. I cannot enable that. With a DB size like mine a full text search takes approx. 10 minutes and kills the server. :-)

**Comment by hoergen** [comment]
> Ok. I guess the full text search has to be enabled for that. I cannot enable that. With a DB size like mine a full text search takes approx. 10 minutes and kills the server. :-)

Ähm ... can't find that option any more. could you please point me to where I should find it?

**Comment by AlfredSK** [comment]
@hoergen  admin -> site -> performance
☑️  Only search in tags 
"On large systems the text search can slow down the system extremely."

**Comment by hoergen** [comment]
Thanks, I found it and the option isn't checked on my server.

**Comment by MrPetovan** [comment]
> To have the possibility to comment on public posts on a foreign profile page is much more easier for the user than going to his/her own instance, copy a link to import a post, to comment and to visit the foreign profile again if there are any other interesting posts.

This is true, but visiting a foreign profile page already is a hit-or-miss depending on the availability of the remote host (try visiting Tobias' profile) that I'm already avoiding it, and this intricate, hard-to-test and hard-to-debug feature wouldn't convince me to do it again. I'm amazed at the OWA feature itself, but I still don't believe in interacting directly on foreign nodes because of the huge potential difference in user experience (performance, theming, node settings).

**Comment by rabuzarus** [comment]
> profile page already is a hit-or-miss depending on the availability of the remote host

Yes but my experience are that over 95% of the profiles I visit are available. And if I want to import a post my server doesn't know (which is nearly every public post beside from the helper/developer forum and 5 developer contacts) I have to visit the foreign profile anyway to search for this post.

The performance issue of import a puplic post from within the local instance would be the same like my approach because it would be nearly the same route but my approach does it all the things in the background

But we would also need an import function which works manually, for importing posts of non-friendica contacts

**Comment by hoergen** [comment]
When we talk about performance we should look at how often this feature would be used. And I t think that this would be a feature that is used here and then and not every minute.

**Comment by tobiasd** [comment]
My node is running 24/7, there is a problem with TLS1.3 and your browser not being able to use a lower version ;-) it is either fixed by the release of `Buster` to Raspbian or me finding time to rebuild the web-server package from Buster and install it on current `Stretch`.

**Comment by clacke** [comment]
> I won't suggest to use the search field, because then you can't search for URLs any more (what is possible at the moment)

You can! Just use the direct hit as the first search result and below that show the full text search results.

**Comment by MrPetovan** [comment]
Han, good idea.
